text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+  names as used elsewhere in the broader ecosystem (such as in other
+  toolchains), unless they have a very good reason to diverge.
+- Tier 3 targets may have unusual requirements to build or use, but must not
+  create legal issues for the Rust project or for developers who work on those
+  targets.
+- Where possible, tier 3 targets may wish to provide documentation for the Rust
+  community for how to build and run tests for the platform, ideally using
+  emulation.
+- Tier 3 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to maintain the target. In particular,
+  do not post comments (automated or manual) on a PR that suggests a block on
+  the PR based on the target. (A PR author may choose to help with a tier 3
+  target but is not required to.)
+- Patches adding or updating tier 3 targets must not break any existing target.
+- If a tier 3 target shows no signs of activity and has not built for some
+  time, and removing it would improve the quality of the Rust codebase, we may
+  post a PR to remove it; any such PR will be CCed to people who have
+  previously worked on the platform, to check potential interest.
+
+## Tier 2 target policy
+
+At this tier, the Rust project guarantees that a target builds, and will reject
+patches that fail to build on a target. Thus, we place requirements that ensure
+the target will not block forward progress of the Rust project.
+
+- Any new tier 2 target requires compiler team approval based on these
+  requirements.
+- In addition, the infrastructure team must approve the integration of the
+  target into CI, and the CI-related requirements. This review and approval
+  will typically take place in the PR adding the target to CI.
+- A tier 2 target must have value to people other than its maintainers.
+- Any new tier 2 target must have a designated team of developers on call to
+  consult on target-specific build-breaking issues, or if necessary to develop
+  target-specific language or library implementation details. (This team should
+  in almost all cases have at least 2 developers.)
+- The target must not place undue burden on Rust developers not specifically
+  concerned with that target. Rust developers may be expected to not
+  gratuitously break a tier 2 target, but are not expected to become experts in
+  every tier 2 target, and are not expected to provide target-specific
+  implementations for every tier 2 target.
+- Where possible, tier 2 targets should provide documentation for the Rust
+  community for how to build and run tests for the platform, ideally using
+  emulation.
+- The target development team should not only fix target-specific issues, but
+  should use any such issue as an opportunity to educate the Rust community
+  about portability to their target, and enhance their documentation of the
+  target.
+- The target must build reliably in CI.
+- Building the target must not take substantially longer than other targets.
```

Comment:
Can we impose a specific restriction, or at least say roughly where the grey area is? E.g., "If the build for a proposed tier 2 target takes more than x% longer than the {median build time | current slowest-building target}, the target development team must demonstrate that the value of the target outweighs the build-time cost."